src/main/java/seedu/address/logic/commands/AddNoteCommand.java

```diff
@@ -0,0 +1,110 @@
+ */
+public class AddNoteCommand extends Command {
+
+    public static final String COMMAND_WORD = "addnote";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the note of the person whose profile matches "
+            + "the specified IC (case-insensitive). "
+            + "Existing remark will be appended by default. To replace the original note, add -replace at "
+            + "the end of your command. E.g. addnote i/S0123456Q n/Diabetes -replace\n"
+            + "Parameters: "
+            + "[" + PREFIX_IC + "IC] "
+            + "[" + PREFIX_NOTE + "NOTE] \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_IC + " S0123456Q "
+            + PREFIX_NOTE + "Healthy.";
+
+
+    public static final String MESSAGE_ADD_NOTE_SUCCESS = "Added note to Person: %1$s";
+    public static final String MESSAGE_DELETE_NOTE_SUCCESS = "Removed note from Person: %1$s";
+    private final IdentityCardNumberMatchesPredicate ic;
+    private final Note note;
+    private final boolean isReplace;
+
+    /**
+     * @param ic of the person in the filtered person list to edit the note
+     * @param note of the person to be updated to
+     */
+    public AddNoteCommand(IdentityCardNumberMatchesPredicate ic, Note note, boolean isReplace) {
+        requireAllNonNull(ic, note);
+
+        this.ic = ic;
+        this.note = note;
+        this.isReplace = isReplace;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        model.updateFilteredPersonList(ic);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (lastShownList.isEmpty()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON);
+        }
+
+        Person personToEdit = lastShownList.get(0);
+        if (isReplace) {
+            Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                    personToEdit.getIdentityCardNumber(), personToEdit.getAge(), personToEdit.getSex(),
+                    personToEdit.getAddress(), note, personToEdit.getTags());
+
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+            return new CommandResult(generateSuccessMessage(editedPerson));
+        } else {
+            Note updatedNote = personToEdit.getNote().append("\n" + note.toString());
+            Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                    personToEdit.getIdentityCardNumber(), personToEdit.getAge(), personToEdit.getSex(),
+                    personToEdit.getAddress(), updatedNote, personToEdit.getTags());
+
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+            return new CommandResult(generateSuccessMessage(editedPerson));
```

Comment:
Would you want to refactor out the repeated lines of code here outside the if else block? You could define `editedPerson` above the if-else block and then move lines 79-82 below the block. I'm not sure if doing that would make readability worse.